lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -494,6 +499,81 @@ def test_delete_submission_scores_attempts_children(self):
         self.assertEqual(unrelated_state['brains'], 'zombie')
 
 
+class TestStudentModuleGrading(SharedModuleStoreTestCase):
+    """
+    Tests the effects of student module manipulations
+    on student grades.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestStudentModuleGrading, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.course_key = cls.course.location.course_key
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            category="chapter",
+            display_name="Test Chapter"
+        )
+        cls.sequence = ItemFactory.create(
+            parent=cls.chapter,
+            category='sequential',
+            display_name="Test Sequential 1",
+            graded=True
+        )
+        cls.vertical = ItemFactory.create(
+            parent=cls.sequence,
+            category='vertical',
+            display_name='Test Vertical 1'
+        )
+        problem_xml = MultipleChoiceResponseXMLFactory().build_xml(
+            question_text='The correct answer is Choice 3',
+            choices=[False, False, True, False],
+            choice_names=['choice_0', 'choice_1', 'choice_2', 'choice_3']
+        )
+        cls.problem = ItemFactory.create(
+            parent=cls.vertical,
+            category="problem",
+            display_name="Test Problem",
+            data=problem_xml
+        )
+        cls.request = get_request_for_user(UserFactory())
+        cls.user = cls.request.user
+
+    def _get_subsection_grade_and_verify(self, all_earned, graded_earned, all_possible, graded_possible):
+        """
+        Retrieves the subsection grade and verifies that
+        its scores match those expected.
+        """
+        subsection_grade_factory = SubsectionGradeFactory(
+            self.user,
+            self.course,
+            get_course_blocks(self.user, self.course.location)
+        )
+        grade = subsection_grade_factory.update(self.sequence)
+        self.assertEqual(grade.all_total.earned, all_earned)
+        self.assertEqual(grade.graded_total.earned, graded_earned)
+        self.assertEqual(grade.all_total.possible, all_possible)
+        self.assertEqual(grade.graded_total.possible, graded_possible)
+
+    @patch('crum.get_current_request')
+    def test_delete_student_state(self, _crum_mock):
+        problem_location = self.problem.location
+        self._get_subsection_grade_and_verify(0, 0, 1, 1)
```

Comment:
Once you update the order of the arguments, it'll be easier to read this as fractions: 0, 1.